project name stays in text field when you go back

emmapainter opened the pull request:
Elevated the state so that the project name can remain filled when the user continues to the next modal then navigates back to the new project view


## Summary

Lifted the project name state from the new project view into the modal container so that the project name persists when you continue to the import media view then navigate back again.

<hr/>

Currently the project name clears when the user navigates back to the project name view, which would be annoying if they only wanted to fix a simple typo.

<hr/>


### OS

<!-- To select your OS. Place an 'x' within [ ]. eg. - [x] Linux -->

- [x] Linux
- [x] MacOS
- [x] Windows

emmapainter committed:
project name stays in text field when you go back

Elevated the state so that the project name can remain filled when the user continues to the next modal then navigates back to the new project view

exquisitebin commented:
I'm currently working on Issue #101, in which the idea is that the user is prompted with a dialog asking if they want to abandon the new project when they click outside of the Modal. Where they can choose to continue and not close the Modal, or close the Modal and start a new project again.

Not sure if remembering the previous project name clashes with this idea. Happy to hear thoughts on this.

exquisitebin reviewed (changes requested):
I'm currently working on Issue https://github.com/chloebrett/mlvet/issues/101 in branch [bugfix/click-outside-of-modal](https://github.com/chloebrett/mlvet/tree/bugfix/click-outside-of-modal). The idea is that the user is prompted with a dialog asking if they want to abandon the new project when they click outside of the Modal. Where they can choose to continue and not close the Modal, or close the Modal and start a new project again.

Not sure if remembering the previous project name clashes with this idea. Happy to hear thoughts on this.

Taz17 commented:
@exquisitebin If you will be showing a dialog and the user confirms they want to exit then you should use the `setProjectName` hook to re-set the `projectName` back to an empty string.

This PR has more to do with how the app behaves when you go back a step in the existing modal rather than what happens when you click off

exquisitebin reviewed (changes requested):
Mutaz clarified my query, and I agree it doesn't clash with the issue I'm working on.

I just have a bug that when you press back to name the new project the continue button is disabled until you change the text (which ideally you wouldn't have to).

![image](https://user-images.githubusercontent.com/58348527/183233806-417ab00a-0c34-4ccb-b483-e17d0662f7f0.png)

fwang1703 reviewed (approved):
Nice!

emmapainter committed:
continue button enable/disable behaviour

Continue button now has correct behaviour when returning to the new project modal. Also streamlined the logic used to determine if the continue button should be disabled.

emmapainter commented:
> Mutaz clarified my query, and I agree it doesn't clash with the issue I'm working on.
> 
> I just have a bug that when you press back to name the new project the continue button is disabled until you change the text (which ideally you wouldn't have to).
> 
> ![image](https://user-images.githubusercontent.com/58348527/183233806-417ab00a-0c34-4ccb-b483-e17d0662f7f0.png)

@exquisitebin this should be fixed now

exquisitebin reviewed (approved):
Looks good, awesome!

rileykeane commented on src/renderer/components/NewProjectModal/NewProjectView.tsx:
```diff
@@ -60,17 +63,12 @@ const NewProjectView = ({ closeModal, nextView }: Props) => {
     <PrimaryButton
       onClick={handleContinue}
-      disabled={isAwaitingProjectName}
+      disabled={!projectName.trim()}
```
👀
